Firestore/core/src/firebase/firestore/util/error_apple.h

```diff
@@ -48,6 +48,22 @@ inline NSError* MakeNSError(const util::Status& status) {
   return MakeNSError(status.code(), status.error_message());
 }
 
+inline util::Status MakeStatus(NSError* error) {
```

Comment:
@wilhuff Gil, I have one question re. this PR -- do you think `error_apple.h` is appropriate for converting `NSError`s to `util::Status`es, or should this function live somewhere under `remote/`?

Reply:
As implemented this is a little weird. By the time we're done we won't be converting from NSErrors with the FIRFirestoreErrorDomain because nothing should be creating them. Are you sure we actually need this?

Another alternative is to modify Status::FromNSError to do this properly: https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/util/status_apple.mm#L27

Reply:
You're right, this isn't too hard to avoid the conversion. I didn't want to touch `FSTRemoteStore` in this PR, but it looks better than the conversion function. Thanks for pointing it out.